lib/internal/quic/core.js

```diff
@@ -2601,59 +2601,45 @@ class QuicStream extends Duplex {
     this._readableState.readingMore = true;
     this.on('pause', streamOnPause);
 
-    // See src/node_quic_stream.h for an explanation
-    // of the initial states for unidirectional streams.
-    if (this.unidirectional) {
-      if (session instanceof QuicServerSession) {
-        if (this.serverInitiated) {
-          // Close the readable side
-          this.push(null);
-          this.read();
-        } else {
-          // Close the writable side
-          this.end();
-        }
-      } else if (this.serverInitiated) {
-        // Close the writable side
-        this.end();
-      } else {
-        this.push(null);
-        this.read();
-      }
-    }
-
     // The QuicStream writes are corked until kSetHandle
     // is set, ensuring that writes are buffered in JavaScript
     // until we have somewhere to send them.
     this.cork();
   }
 
-  // Set handle is called once the QuicSession has been able
-  // to complete creation of the internal QuicStream handle.
-  // This will happen only after the QuicSession's own
-  // internal handle has been created. The QuicStream object
-  // is still minimally usable before this but any data
-  // written will be buffered until kSetHandle is called.
-  [kSetHandle](handle) {
-    this[kHandle] = handle;
-    if (handle !== undefined) {
-      handle.onread = onStreamRead;
-      handle[owner_symbol] = this;
-      this[async_id_symbol] = handle.getAsyncId();
-      this.#id = handle.id();
-      this.#dataRateHistogram = new Histogram(handle.rate);
-      this.#dataSizeHistogram = new Histogram(handle.size);
-      this.#dataAckHistogram = new Histogram(handle.ack);
-      this.uncork();
-      this.emit('ready');
-    } else {
-      if (this.#dataRateHistogram)
-        this.#dataRateHistogram[kDestroyHistogram]();
-      if (this.#dataSizeHistogram)
-        this.#dataSizeHistogram[kDestroyHistogram]();
-      if (this.#dataAckHistogram)
-        this.#dataAckHistogram[kDestroyHistogram]();
-    }
+  _construct(cb) {
+    // Set handle is called once the QuicSession has been able
+    // to complete creation of the internal QuicStream handle.
+    // This will happen only after the QuicSession's own
+    // internal handle has been created. The QuicStream object
+    // is still minimally usable before this but any data
+    // written will be buffered until kSetHandle is called.
+    this[kSetHandle] = (handle) => {
```

Comment:
I really dislike the property pattern for these functions. I'd rather `[kSetHandle]` be a function on the class.